DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -1,244 +1,286 @@
-
-  float get_iRinv();
-  float get_iphi();
-  float get_itanl();
-  float get_iz0();
-  float get_id0();
-  float get_ichi2XY();
-  float get_ichi2Z();
-  float get_iBendChi2();
-
-  // separate functions for unpacking 96-bit track word
-
-  float unpack_iRinv();
-  float unpack_iphi();
-  float unpack_itanl();
-  float unpack_iz0();
-  float unpack_id0();
-  float unpack_ichi2XY();
-  float unpack_ichi2Z();
-  float unpack_iBendChi2();
-  unsigned int unpack_ispare();
-  unsigned int unpack_hitPattern();
-
-  // getters for packed bits.
-  // These functions return, literally, the packed bits in integer format for each quantity.
-  // Signed quantities have the sign enconded in the left-most bit.
+                    unsigned int chi2RPhi,  // would be total chisq if chi2Z is zero
+                    unsigned int chi2RZ,
+                    unsigned int bendChi2,
+                    unsigned int hitPattern,
+                    unsigned int mvaQuality,
+                    unsigned int mvaOther);
+
+  void setTrackWord(ap_uint<TrackBitWidths::kValidSize> valid,
+                    ap_uint<TrackBitWidths::kRinvSize> rInv,
+                    ap_uint<TrackBitWidths::kPhiSize> phi0,
+                    ap_uint<TrackBitWidths::kTanlSize> tanl,
+                    ap_uint<TrackBitWidths::kZ0Size> z0,
+                    ap_uint<TrackBitWidths::kD0Size> d0,
+                    ap_uint<TrackBitWidths::kChi2RPhiSize> chi2RPhi,  // would be total chisq if chi2Z is zero
+                    ap_uint<TrackBitWidths::kChi2RZSize> chi2RZ,
+                    ap_uint<TrackBitWidths::kBendChi2Size> bendChi2,
+                    ap_uint<TrackBitWidths::kHitPatternSize> hitPattern,
+                    ap_uint<TrackBitWidths::kMVAQualitySize> mvaQuality,
+                    ap_uint<TrackBitWidths::kMVAOtherSize> mvaOther);
 
-  unsigned int get_hitPattern();
-  unsigned int get_ispare();  // will need a converter(s) when spare bits are defined
-
-  unsigned int get_RinvBits();
-  unsigned int get_phiBits();
-  unsigned int get_tanlBits();
-  unsigned int get_z0Bits();
-  unsigned int get_d0Bits();
-  unsigned int get_chi2XYBits();
-  unsigned int get_chi2ZBits();
-  unsigned int get_BendChi2Bits();
-
-  // copy constructor
-
-  TTTrack_TrackWord(const TTTrack_TrackWord& word) {
-    initialize();
-
-    iRinv = word.iRinv;
-    iphi = word.iphi;
-    itanl = word.itanl;
-    iz0 = word.iz0;
-    id0 = word.id0;
-    ichi2XY = word.ichi2XY;
-    ichi2Z = word.ichi2Z;
-    iBendChi2 = word.iBendChi2;
-    ispare = word.ispare;
-    iHitPattern = word.iHitPattern;
-
-    // three 32-bit packed words
-    TrackWord1 = word.TrackWord1;
-    TrackWord2 = word.TrackWord2;
-    TrackWord3 = word.TrackWord3;
+private:
+  // ----------private member functions --------------
+  unsigned int digitizeSignedValue(double value, unsigned int nBits, double lsb) const {
+    unsigned int digitized_value = std::floor(std::abs(value) / lsb);
```

Comment:
I think the correct eqns. are:
int J = floor(value/lsb); 
unsigned bits = (value > 0) ? J : pow(2,nBits) + J.

Reply:
just to avoid someone to copy-paste and trip on the same mistake (as I just did ☹️), it should  be `>= 0` and not `> 0` .